fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/mixin/networking/CustomPayloadC2SPacketMixin.java

```diff
@@ -28,21 +26,18 @@
 import net.minecraft.network.packet.c2s.common.CustomPayloadC2SPacket;
 import net.minecraft.util.Identifier;
 
-import net.fabricmc.fabric.impl.networking.payload.PacketByteBufPayload;
-import net.fabricmc.fabric.impl.networking.payload.PayloadHelper;
+import net.fabricmc.fabric.api.networking.v1.PacketByteBufs;
+import net.fabricmc.fabric.impl.networking.payload.RetainedPayload;
 
 @Mixin(CustomPayloadC2SPacket.class)
 public class CustomPayloadC2SPacketMixin {
-	@Shadow
-	@Final
-	private static int MAX_PAYLOAD_SIZE;
```

Comment:
We still need to check the max payload size, in the same way that vanilla does. I previously moved this logic to `PayloadHelper.read`.  It may be bestto just copy the vanilla behvaiour into each of these mixins.